### PR TITLE
Grand Prosthetics Rework [MERGE ME PLEASE GOD]

### PR DESCRIPTION
**Pull request by VaelophisNyx**
## About The Pull Request

Played a Kepori with Prosthetic Limb as a quirk. Discovered that they just cant have base prosthetics. This fixes that, and more.

Adds surplus prosthetics to the (medical) protolathe Medical Designs tab as well as to the exosuit Misc tab.
Makes surplus prosthetic designs unlocked at Biological Technology
Removed the Random prosthetic quirk
added buttons under Character Appearance to allow manual choice of amputations or prosthetics
Adds a set of surplus prosthetics specifically for Vox, and one for Kepori, so they can properly have a prosthetic limb without it popping off of them.
Adds the new surplus prosthetics for kepori and vox to the surplus prosthetic crate pre-fill.
Adds Vox prosthetic sprites (thank you very much @drawsstuff)
Had to touch IPC species code to fix some runtimes, but the fix was insanely simple

## Why It's Good For The Game

Vox & Kepori can now safely take Prosthetic Limbs at round start and not start missing a limb entirely. Additionally you can now print cheap (3 sheets of iron) surplus prosthetics from a protolathe or exosuit fab, for if you need a full limb replacement and can't get a compatible organic limb in a timely manner.

## Changelog
:cl: VaelophisNyx, @Drawsstuff, @That0nePerson, @Chubbygummibear, @MarkSuckerberg 
add: Adds surplus prosthetics to the protolathe Medical Designs tab , and exosuit fabricator Misc tab. 3000 units of iron each.
add: Adds a set of surplus prosthetics specifically for Kepori, and one set for Vox.
add: Adds the new surplus prosthetics for Kepori and Vox to the surplus prosthetic crate pre-fill.
add: Adds new Prosthetic Limbs character appearance customization options
tweak: Makes surplus prosthetic designs unlocked at Biological Technology.
remove: Removed the random Prosthetic Limb quirk. Please reset quirks to remove it from any characters that retain it. It does nothing anymore.
/:cl:

### CODE REVIEW

**Comment by VaelophisNyx** [comment]
whoops definitely didnt forget to fix the design names

**Comment by VaelophisNyx** [comment]
going to push the full rework to this and update the PR; may as well, I have the time.

**Comment by VaelophisNyx** [comment]
A lot going on here now, but I'm pushing to get assistance with the buttons to choose your amputations/prosthetics. They just....aren't working at all, and I cannot figure out why

**Comment by VaelophisNyx** [comment]
Turns out I was wrong, it just works actually.
I just forgot  to reset my character saves so it was breaking with old data

**Review by triplezeta — APPROVED** [review]
No particular issues from me spritewise (I would like to mention that Piper is currently working on prosthetic resprites for Kepori, but we can get those in a later PR). 

I believe there might be an issue with the changelog (unsure if we use `rscadd` and `rscdel`). Consider rewriting it? 

I'm also unsure of having three different sets of prosthetic limbs, but I will let someone else who is more familiar with limbcode comment on that.

**Comment by VaelophisNyx** [comment]
Unfortunately this was the only way I knew how to do it, otherwise I'd've done a sprite-swapping system with one set

**Comment by Zevotech** [comment]
@zestybastard your dream PR is here. no more sawing off the leg roundstart bestie.

**Comment by triplezeta** [comment]
Aye, fair enough. The only other thing I could think of that might be applicable are augments/cyborg limbs, and I'm not even sure those work.

**Comment by VaelophisNyx** [comment]
You would be correct as they have an incorrect body type; I'd need a unique set of cyborg limbs for Vox and Kepori, and that'd get messy very very fast

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -2367,6 +2385,30 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	if("tail_lizard" in pref_species.default_features)
 		character.dna.species.mutant_bodyparts |= "tail_lizard"
 
+	for(var/L in prosthetic_limbs)
```
Comment (Fikou) [inline code/modules/client/preferences.dm]: Use descript variable names

File: code/modules/mob/living/carbon/human/species_types/kepori.dm [hunk 2]
```diff
@@ -41,6 +41,9 @@
 	/// See: [/datum/component/tackler/var/skill_mod]
 	var/skill_mod = 2
 
+	// Does this species have unique robotic limbs? (currently used in: kepori and vox)
```
Comment (Fikou) [inline code/modules/mob/living/carbon/human/species_types/kepori.dm]: why are you adding this comment to everything

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 3]
```diff
@@ -62,6 +62,87 @@
...
 	category = list("Cyborg")
 
+//Prosthetics
+/datum/design/prosthetic_l_arm
+	name = "Surplus Prosthetic Left Arm"
+	id = "prosthetic_l_arm"
+	build_type = PROTOLATHE | MECHFAB
+	build_path = /obj/item/bodypart/l_arm/robot/surplus
+	materials = list(/datum/material/iron=3000)
+	construction_time = 50
```
Comment (Fikou) [inline code/modules/research/designs/mechfabricator_designs.dm]: use SECONDS defines

File: code/modules/client/preferences.dm (hunk 1 above)
Comment (VaelophisNyx) [reply code/modules/client/preferences.dm]: I forgot to update it from the prior person's PR. Not sure what to use though, but I'll look into it

File: code/modules/mob/living/carbon/human/species_types/kepori.dm (hunk 2 above)
Comment (VaelophisNyx) [reply code/modules/mob/living/carbon/human/species_types/kepori.dm]: Clarity more than anything; as well as it being mostly a leftover from the old PR. I don't see an issue with it sticking around as it is only a code comment

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 3 above)
Comment (VaelophisNyx) [reply code/modules/research/designs/mechfabricator_designs.dm]: SECONDS defines?

**Comment by VaelophisNyx** [comment]
gee gitbot it'd be really nice to know WHAT PR the conflict is with

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 3 above)
Comment (triplezeta) [reply code/modules/research/designs/mechfabricator_designs.dm]: `5 SECONDS` instead of `50`

Comment (VaelophisNyx) [reply code/modules/research/designs/mechfabricator_designs.dm]: I will do this but would like to point out not a single other mechfab construction time uses that format.

File: code/modules/client/preferences.dm (hunk 1 above)
Comment (MarkSuckerberg) [reply code/modules/client/preferences.dm]: even just something like `var/limb` would be better

File: code/modules/mob/living/carbon/human/species_types/kepori.dm (hunk 2 above)
Comment (Fikou) [reply code/modules/mob/living/carbon/human/species_types/kepori.dm]: its fine to be on the base type, but it doesnt need to exist on the subtypes
if anyone wants to know what it is they can just click on it with vscode and be linked to the original variable on /datum/species

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 3 above)
Comment (Fikou) [reply code/modules/research/designs/mechfabricator_designs.dm]: thats because they are old as fuck

**Comment by VaelophisNyx** [comment]
alright ffs
I need someone else to look at this and force it to not conflict. it breaks if it doesn't have the regex changed to 121 from 119, but if I do that, it whines about a conflict that shouldn't be possible since it's already pushed to the master branch

**Comment by Fikou** [comment]
> alright ffs I need someone else to look at this and force it to not conflict. it breaks if it doesn't have the regex changed to 121 from 119, but if I do that, it whines about a conflict that shouldn't be possible since it's already pushed to the master branch

your pr changes the amount of text2path uses
some other pr that got merged also changed the amount of text2path uses
so your pr conflicts with that and you need to update the number to the correct one

**Comment by VaelophisNyx** [comment]
Ok I see that but the issue is, that one already got merged and this one increases the number to work on top of that change. I do not see why this is a *conflict*
if the number doesn't conflict this fails to compile since it requires +2 higher text2path uses than the master branch.

**Comment by Fikou** [comment]
> Ok I see that but the issue is, that one already got merged and this one increases the number to work on top of that change. I do not see why this is a _conflict_ if the number doesn't conflict this fails to compile since it requires +2 higher text2path uses than the master branch.

just delete the merge conflict markers then

**Comment by VaelophisNyx** [comment]
I'm not sure I actually can, as I don't have write access to this repo, and I'm pretty sure that requires write access to do.

**Comment by VaelophisNyx** [comment]
Or...not and the UI is just incredibly user unfriendly

**Comment by Fikou** [comment]
> I'm not sure I actually can, as I don't have write access to this repo, and I'm pretty sure that requires write access to do.

you are editing your own version of the repo with updates from us, not our repo

**Comment by Zevotech** [comment]
> added buttons under Character Appearance to allow manual choice of amputations or prosthetics

does this allow for full body prosthetics? or just limbs?

**Comment by Ms-Mee** [comment]
> > added buttons under Character Appearance to allow manual choice of amputations or prosthetics
> 
> does this allow for full body prosthetics? or just limbs?

Just limbs

**Comment by Sylphily** [comment]
did you test even test the PR?
half the shit is broken
can not select limbs and randomly adds the priview of limbs to the character page

**Comment by VaelophisNyx** [comment]
Yes I did indeed test it, but not on normal humanoids, just vox and kepori. I am actively working on it, something fucked up between finishing this and testmerging it

**Comment by VaelophisNyx** [comment]
This is ready to go now, shouldn't have any issues remaining and even fixes some IPC runtimes

File: code/modules/client/preferences.dm [hunk 4]
```diff
@@ -2033,6 +2044,13 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 						facial_hairstyle = random_facial_hairstyle(gender)
 						hairstyle = random_hairstyle(gender)
 
+				if("limbs")
+					if(href_list["customize_limb"])
+						var/limb = href_list["customize_limb"]
+						var/status = input(user, "You are modifying your [parse_zone(limb)], what should it be changed to?", "Character Preference", prosthetic_limbs[limb]) as null|anything in list(PROSTHETIC_NORMAL,PROSTHETIC_ROBOTIC,PROSTHETIC_AMPUTATED)
```
Comment (Bokkiewokkie) [inline code/modules/client/preferences.dm:2117]: `list(PROSTHETIC_NORMAL,PROSTHETIC_ROBOTIC,PROSTHETIC_AMPUTATED)`
Should be a global list initialized in flavor_misc.dm

**Review by Bokkiewokkie — CHANGES_REQUESTED** [review]

**Comment by Skies-Of-Blue** [comment]
As an aside, how problematic would it be to keep the random prosthetic quirk in addition to the new system? I personally have a couple of characters that are built around having a new prosthetic every round. While it's not too difficult to select a random one before every round, it IS a nice QoL to have that done automatically.

**Comment by VaelophisNyx** [comment]
I was asked to remove the quirk so, I'd ask in the development chat if anyone would care if it remained.

File: code/modules/client/preferences.dm (hunk 4 above)
Comment (VaelophisNyx) [reply code/modules/client/preferences.dm:2117]: You sure? The way it is works, at least on my end.

**Comment by VaelophisNyx** [comment]
I don't know if I can fix this without help from @/MarkSuckerberg and I really don't want to bother him because of how busy he is

**Comment by CenkTheRaptor** [comment]
kepori augments are completely broken. it gives the wrong type and they look like they're floating because of it

**Comment by VaelophisNyx** [comment]
they do? Do you have steps to reproduce this?
also AUGMENTS or PROSTHETICS
that matters

**Comment by CenkTheRaptor** [comment]
simply selecting it to be prosthetic causes the issue, idk. for some reason saving the setup changes it to the proper sprite, but you don't spawn with it, it spawns with a generic one that doesn't fit

**Comment by tiramisuapimancer** [comment]
Can you pretty please make it so that having prosthetics/amputations grants a quirk point? It's bound to screw with existing character's setups. You can check Citadel's implementation of this exact thing for how they did it, if you need me to go looking for that PR?

**Comment by CenkTheRaptor** [comment]
> simply selecting it to be prosthetic causes the issue, idk. for some reason saving the setup changes it to the proper sprite, but you don't spawn with it, it spawns with a generic one that doesn't fit

it seems to randomly change upon saving the setup.
https://gyazo.com/9d3a6de2653387173b31b50099dbd763
ingame it has the wrong sprite. https://gyazo.com/139194fb3b7f737df905a01f5412a481

**Comment by VaelophisNyx** [comment]
> Can you pretty please make it so that having prosthetics/amputations grants a quirk point? It's bound to screw with existing character's setups. You can check Citadel's implementation of this exact thing for how they did it, if you need me to go looking for that PR?

I'd take that up with whomever controls balance for shiptest; I personally dont see the issue with it not granting points

**Comment by tiramisuapimancer** [comment]
I theoretically myself have the ability to control balance for Shiptest, being a lead, and I would say keeping the balance as it is would be correct. Like I said, it breaks existing character slots and forces people to rearrange their quirk balances. You're technically changing balance by making it not give a point.

**Comment by VaelophisNyx** [comment]
dear maptainers: there should be no actual map changes, github desktop just *refused* to let me discard a pulled map.
the file is actually identical to the master version as well.

**Comment by ZephyrTFA** [comment]
> dear maptainers: there should be no actual map changes, github desktop just _refused_ to let me discard a pulled map. the file is actually identical to the master version as well.

This is because you are changing the lineendings per the git hook in the repo, this is fine

**Comment by VaelophisNyx** [comment]
I didn't even touch the file
I hit pull
and it just
did that on its own

**Pull request merged**